misc/ls-extensions/modules/json-to-record-converter/src/test/java/io/ballerina/converters/JsonToRecordConverterTests.java

```diff
@@ -0,0 +1,176 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Tests for JsonToRecordConverter.
+ */
+public class JsonToRecordConverterTests {
+    private static final Path RES_DIR = Paths.get("src/test/resources/").toAbsolutePath();
+    ArrayList<TypeDefinitionNode> records;
+
+    private final Path basicSchemaJson = RES_DIR.resolve("json")
+            .resolve("basic_schema.json");
+    private final Path basicSchemaBal = RES_DIR.resolve("ballerina")
+            .resolve("basic_schema.bal");
+
+    private final Path basicObjectJson = RES_DIR.resolve("json")
+            .resolve("basic_object.json");
+    private final Path basicObjectBal = RES_DIR.resolve("ballerina")
+            .resolve("basic_object.bal");
+
+    private final Path nestedSchemaJson = RES_DIR.resolve("json")
+            .resolve("nested_schema.json");
+    private final Path nestedSchemaBal = RES_DIR.resolve("ballerina")
+            .resolve("nested_schema.bal");
+
+    private final Path nestedObjectJson = RES_DIR.resolve("json")
+            .resolve("nested_object.json");
+    private final Path nestedObjectBal = RES_DIR.resolve("ballerina")
+            .resolve("nested_object.bal");
+
+    private final Path sample1Json = RES_DIR.resolve("json")
+            .resolve("sample_1.json");
+    private final Path sample1Bal = RES_DIR.resolve("ballerina")
+            .resolve("sample_1.bal");
+
+    private final Path sample2Json = RES_DIR.resolve("json")
+            .resolve("sample_2.json");
+    private final Path sample2Bal = RES_DIR.resolve("ballerina")
+            .resolve("sample_2.bal");
+
+    private final Path sample3Json = RES_DIR.resolve("json")
+            .resolve("sample_3.json");
+    private final Path sample3Bal = RES_DIR.resolve("ballerina")
+            .resolve("sample_3.bal");
+
+    private final Path sample4Json = RES_DIR.resolve("json")
+            .resolve("sample_4.json");
+    private final Path sample4Bal = RES_DIR.resolve("ballerina")
+            .resolve("sample_4.bal");
+
+    private final Path sample5Json = RES_DIR.resolve("json")
+            .resolve("sample_5.json");
+    private final Path sample5Bal = RES_DIR.resolve("ballerina")
+            .resolve("sample_5.bal");
+
+    private final Path sample6Json = RES_DIR.resolve("json")
+            .resolve("sample_6.json");
+    private final Path sample6Bal = RES_DIR.resolve("ballerina")
+            .resolve("sample_6.bal");
+
+    private final Path crlfJson = RES_DIR.resolve("json")
+            .resolve("crlf.json");
+    private final Path crlfBal = RES_DIR.resolve("ballerina")
+            .resolve("from_crlf.bal");
+
+    private final Path emptyBal = RES_DIR.resolve("ballerina")
+            .resolve("empty.bal");
+
+    // write the codeblock to an empty .bal file
+    public void writeToEmpty(String codeBlock) {
```

Comment:
This looks like an unused function. Is it kept intentionally?

Reply:
It was kept as it might be useful when adding new tests and checking outputs. Can remove as it doesn't serve a meaningful purpose

Reply:
removed in 3459367